docs/contributing.md

```diff
@@ -35,6 +35,38 @@ Adding examples is quite straight forward, so don't feel shy! Here's how to do i
+
+To make it easier to contribute a new generator, and to avoid focusing too much of the internals of Modelina, we created a template generator to get you started. If you encounter discreprencies with the following guide or templates, make sure to raise it as an issue so it can be fixed!
+
+Here is how:
+1. Copy/paste the [template generator](../src/generators/template/) and [tests](../test/generators/template/) and rename it to your generator.
+2. Search and replace within your new generator and test folder for `Template`, `template` and `TEMPLATE` and replace it with your generator name and match the cases. **Make sure you search and replace it with matching case**.
+3. Replace the filenames `Template...` with your generator name.
+4. Add your [generator to the generator index file](../src/generators/index.ts).
+
+Now it's time to adapt the template into what ever it is you are generating:
+1. Adapt the [constraint logic](../src/generators/template/) based on what is allowed within your output. Read more about the constraint logic here.
+2. Add all of the reserved keywords that the models must never generate in the [Constant file](../src/generators/template/Constants.ts).
+3. Adapt/create the first renderers. The template by default include two renderers, one for rendering enums and one for classes, but you can create what ever renderers makes sense. For example in Rust it's not called class but struct, so therefore its called a `StructRenderer`.
+4. Adapt the file generator and the rendering of the complete models to fit your generator.
+
+Time to adapt the tests, cause without tests, it's just an empty promise. The test that is included in the template is really just placeholders, so make sure you adapt them accordingly to your code.
+1. Add a mocked renderer in the [TestRenderers](../test/TestUtils/TestRenderers.ts) file.
+2. Adapt the [constrainer tests](../test/generators/template/TemplateConstrainer.spec.ts) based on the output.
+3. Adapt the [reserved keywords tests](../test/generators/template/Constants.spec.ts)
+4. Add your generator to the [FileGenerators test](../test/generators/FileGenerators.spec.ts) to ensure the models are accurately written to files.
+
+Lastly, we need to adapt some of the docs to showcase your new awesome generator! Cause if the users cant find it, it dont exist.
+1. Add your [generator specific documentation under languages](./languages/) and add it to the [list of generators](./README.md#languages)
+2. Add your generator to the list of generators in the [main readme file](../README.md)
+3. Add a basic [usage example to the usage documentation](./usage.md), you can see more about how to create [examples here](#adding-examples).
+
+Aaaand that's it! As a rule of thumb, start small and slowly add more features, dont try to push everything into one PR, as it will take forever to review, code, and merge. 
+
+PR's you can look to for guidance on how the process goes: https://github.com/asyncapi/modelina/pull/818
```

Comment:
I wonder if Presets deserve a section here? When I was diving in, I had a hard time figuring out what to put in a "base" Preset and the interplay with Preset overrides. Going through the Java generator helped me understand the abstraction level, so that's a great example to link!

Reply:
That makes sense, adding that 👍

Reply:
I added the following:

```
An important note about presets, they are used to extend and build upon the default model, the bare minimum of a data model, so that Modelina can support multiple features. You can read more about [presets here](./presets.md). If you have any questions or want something clarified, don't hesitate to [reach out on slack](https://asyncapi.com/slack-invite).
```

Let me know if we need to add anything else 🤔